documentation/src/docs/wire-protocol.md

```diff
@@ -7,106 +7,103 @@ You may obtain a copy of the License at
+2.  Offset that was read from: Long (8 bytes).
+3.  Is at Tail: Boolean (1 bit).
+4.  Is at `EndOfSegment`: 1 bit.
+5.  `Data`: Binary (remaining length in the message).
 
-### Segment Read - Reply
+The client requests to read from a particular Segment at a particular Offset. It then receives one or more replies in the form of `SegmentRead` messages. These contain the data they requested (assuming it exists). The server decides on more or less data to give the client than it asked for, in as many replies as it sees fit.
 
-1.  Segment that was read - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
-2.  Offset that was read from - Long (8 bytes)
-3.  Is at Tail - Boolean (1 bit)
-4.  Is at EndOfSegment - (1 bit)
-5.  Data - Binary (remaining length in message)
+# Appending
 
-The client requests to read from a particular stream at a particular offset, it then receives one or more replies in the form of SegmentRead messages. These contain the data they requested (assuming it exists). The server may decided to give the client more or less data than it asked for, in as many replies as it sees fit.
+## Setup Append - Request
+1.  `RequestId`: Long (8 bytes). (This field contains the client-generated ID that has been propagated to identify a client request).
+2.  `writerId`: UUID (16 bytes) identifies the requesting appender.
+3.  Segment to append: String (2 byte length, followed by that many bytes of Java's Modified UTF-8).
+4. `delegationToken`: This was added to perform _auth_. It is an opaque-to-the-client token provided by the Controller that says it's allowed to make this call.
 
-Appending
----------
+## Append Setup - Reply
 
-### Setup Append - Request
+1.  `RequestId`: 8 byte Long. (This field contains the client-generated ID that has been propagated to identify a client request).
+2.  Segment to append: String (2 byte length, followed by that many bytes of Java's Modified UTF-8).
+3.  `writerId`: UUID (16 bytes) identifies the requesting appender.
+4.  `lastEventNumber`: Long (8 bytes). Specifies the last event number in the Stream.
 
-1.  ConnectionId - UUID (16 bytes) Identifies this appender.
-2.  Segment to append to. - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
+## AppendBlock - Request
 
-### Append Setup - Reply
+1. `writerId`: UUID (16 bytes) identifies the requesting appender.
+2. `Data`: This holds the contents of the block.
 
-1.  Segment that can be appended to. - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
-2.  ConnectionId - UUID (16 bytes) Identifies the requesting appender.
-3.  ConnectionOffsetAckLevel - Long (8 bytes) What was the last offset received and stored on this segment for this connectionId (0 if new)
+## AppendBlockEnd - Request
 
-### BeginAppendBlock - Request 
+1. `writerId`: UUID (16 bytes) identifies the requesting appender.
+2. `sizeOfWholeEvents`: Integer (4 bytes).
```

Comment:
Any description for 'sizeOfWholeEvents`

Reply:
can i include the following:
 'sizeOfWholeEvents`  :  Allows the append block to be less than full

Reply:
@tkaitchuck please address thew ablove comments

Reply:
It is the total number of bytes in this block (starting from the beginning) that is composed of whole (meaning non-partial) events.

Reply:
Accepted suggestion